core/src/main/resources/db/changelog/changesets/2024_01_22_survey_fields.yaml

```diff
@@ -0,0 +1,52 @@
+databaseChangeLog:
+  - changeSet:
+      id: "survey_fields"
+      author: dbush
+      changes:
+        - addColumn:
+            tableName: survey
+            columns:
+              - column: { name: eligibility_rule, type: text }
+              - column: { name: required, type: boolean, defaultValueBoolean: false, constraints: { nullable: false } }
+              - column: { name: allow_admin_edit, type: boolean, defaultValueBoolean: true, constraints: { nullable: false } }
+              - column: { name: allow_participant_start, type: boolean, defaultValueBoolean: true, constraints: { nullable: false } }
+              - column: { name: allow_participant_reedit, type: boolean, defaultValueBoolean: true, constraints: { nullable: false } }
+              - column: { name: prepopulate, type: boolean, defaultValueBoolean: false, constraints: { nullable: false } }
+              - column: { name: recur, type: boolean, defaultValueBoolean: false, constraints: { nullable: false } }
+              - column: { name: recurrence_interval_days, type: integer }
+              - column: { name: days_after_eligible, type: integer }
+        - sql:
+            sql: |
+              update survey set 
+                eligibility_rule = study_environment_survey.eligibility_rule, 
+                allow_admin_edit = study_environment_survey.allow_admin_edit, 
+                allow_participant_start = study_environment_survey.allow_participant_start, 
+                allow_participant_reedit = study_environment_survey.allow_participant_reedit,
+                prepopulate = study_environment_survey.prepopulate, 
+                recur = study_environment_survey.recur, 
+                recurrence_interval_days = study_environment_survey.recurrence_interval_days, 
+                days_after_eligible = study_environment_survey.days_after_eligible, 
+                required = study_environment_survey.required
+              from study_environment_survey
+              where survey.id = study_environment_survey.survey_id;
+        - dropColumn:
```

Comment:
is there a good way to make ensure that this works? like, test it on the dev database or something? also, do we need to check in `study_envrionment_survey` that the SES is still active? i just had an experience from duos where we had a complex changeset like this totally break stuff and i am so scared now haha

Reply:
yeah, this script will run when you checked out the branch for local testing, and will also run when it gets merged to demo, so that will be our test.  (I also ran it locally a few times).  Theoretically we shouldn't need to check that the SES is still active -- we want this script to run on both active and inactive SES's.  The inactive ones are needed to update older versions of the surveys with correct configs